### PR TITLE
[Modals] Enable closing modal by pressing “Esc” and handle focus of inputs inside the modal

### LINKED ISSUE #768: UX issues with the modal component
2 issues:

* When a modal is opened with one or more input fields, the first input should automatically be focused.
* When the modal or one of its input fields has focus, pressing the Esc key should close the modal

### PR DESCRIPTION
**Pull request by fernandoabolafio**
@sndurkin regarding the first issue you mentioned, I couldn't find any modal where the input wasn't being focused. Is there any specific modal where you saw that?
closes #768

### CODE REVIEW

**Comment by fernandoabolafio** [comment]
@sndurkin I addressed your comments by focusing the first input shown in the modal once the component has been mounted.

File: src/components/Modal/ModalContentWrapper.js [hunk 1]
```diff
@@ -19,32 +21,79 @@ const SubmitButton = ({ onClick, text, disabled }) => (
...
-          {onClose && <span style={{ fontSize: "18px", cursor: "pointer" }} onClick={onClose}>✖</span>}
+class ModalContentWrapper extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      modalID: uniqueID("modal")
+    };
+  }
+  handleKeyUp = (e) => {
+    if (e.keyCode === 27 && !this.props.disableCloseOnEsc) {
```
Comment (brunobraga95) [inline src/components/Modal/ModalContentWrapper.js]: could you move 27 to a constant and give it a meaningful name? like ESC_KEY for example

Comment (brunobraga95) [reply src/components/Modal/ModalContentWrapper.js]: also I believe.

```
const pressedEscKey = e.keyCode === ESC_KEY && !this.props.disableCloseOnEsc
const action = this.props.onClose || this.props.onCancel;
if(pressedEsc && action)
   action()
```

is dryer and cleaner.

File: src/components/Modal/ModalContentWrapper.js [hunk 2]
```diff
@@ -19,32 +21,79 @@ const SubmitButton = ({ onClick, text, disabled }) => (
...
-        {onCancel && <CancelButton onClick={onCancel} text={cancelText} />}
-        {onSubmit && <SubmitButton onClick={onSubmit} text={submitText} disabled={submitDisabled} />}
-      </div>}
-    </div>
-  );
+    );
+  }
+}
+
+ModalContentWrapper.propTypes = {
```
Comment (brunobraga95) [inline src/components/Modal/ModalContentWrapper.js]: :+1:

**Comment by fernandoabolafio** [comment]
thanks for reviewing it @brunobraga95. I addressed your comments.

**Pull request merged**